### PR TITLE
Added ability to filter on 'region', new column in objs table

### PR DESCRIPTION
**Pull request by savage13**
- beco.ts file is new, parses and computes a region
- Added region to objs table
- New option for EcoSystem files (-e ../botw/EcoSystem)
- Tower Names are hard coded into build.ts 
- can use region:region_name is searches

I probably messed up the formatting again.

---
This change is [Reviewable](https://reviewable.io/reviews/zeldamods/radar/6)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,6 @@ A server for querying placement objects in *The Legend of Zelda: Breath of the W
 
 Run build.ts to generate a map database before starting the server for the first time.
 
-    ts-node build.ts -a ../botw/Actor
+    ts-node build.ts -a ../botw/Actor -e ../botw/EcoSystem
```
Comment (leoetlino) [inline README.md:6]: IMO we should just ship the beco files (and put them in content/ecosystem/) -- there are only like 3-4 beco files and they're all tiny.

File: beco.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    posz = clamp(posz, -4000, 4000);
+    let epsx = (posx + 5000 < 0.0) ? -0.5 : 0.5;
+    let epsz = (posz + 4000 < 0.0) ? -0.5 : 0.5;
+    let x = Math.trunc(posx + 5000.0 + epsx);
+    let z = Math.trunc((posz + 4000.0 + epsz) / this.divisor);
+
+    let row = clamp(z, 0, this.num_rows-1);
+    if (this.divisor == 10) {
+      // the hex value should be a long long int
+      x = x / 10 + ((0x66666667 * x) >> 0x3F);
```
Comment (leoetlino) [inline beco.ts:82]: This is just a division by 10, see https://github.com/zeldaret/botw/blob/c42f6cd657220481636165dc1dec63e22e79bc34/src/KingSystem/Ecosystem/ecoSystem.cpp#L44

**Comment by leoetlino** [comment]
> I probably messed up the formatting again.

If you are using VSCode's TypeScript extension, you could run the built-in formatter (in a separate commit please)

**Comment by savage13** [comment]
- Moved beco files into content/ecosystem
- Removed associated command line argument

I will look into the Typescript formatter.

**Comment by savage13** [comment]
Got the Typescript formatter working, but was going to wait to format until this PR was handled.

**Comment by leoetlino** [comment]
Oh, I was waiting for the reformatting of beco.ts (since that's an entirely new file) :sweat_smile:

(Don't bother reformatting the other files; that should probably be done in another PR.)

**Comment by savage13** [comment]
beco.ts formatted :D

**Review by leoetlino — APPROVED** [review]

**Pull request merged**